include/xrpl/protocol/Book.h

```diff
@@ -36,12 +37,17 @@ class Book final : public CountedObject<Book>
 public:
     Issue in;
     Issue out;
+    std::optional<uint256> domain;
 
     Book()
     {
     }
 
-    Book(Issue const& in_, Issue const& out_) : in(in_), out(out_)
+    Book(
+        Issue const& in_,
+        Issue const& out_,
+        std::optional<uint256> const& domain_ = std::nullopt)
```

Comment:
Hi, I think it's better not to have last argument as `std::nullopt` by default - it makes it so easy to forget to pass domain, if it is set.
I would rather ask everyone to pass `std::nullopt` explicitly than forget it once.

Reply:
Thanks for the feedback! fortunately it affected only the deprecated RPC, but I agree we should be on the safe side.  fixed in https://github.com/XRPLF/rippled/pull/5404/commits/5cc46f94beff25caff1973fa069aa0620886a07b

Reply:
Thanks!